src/main/java/com/backend/controller/AssetRefDataController.java

```diff
@@ -0,0 +1,79 @@
+package com.backend.controller;
+
+import java.time.LocalDate;
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.MediaType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import com.backend.model.AssetRefData;
+import com.backend.configuration.Constants;
+import com.backend.exception.BadAssetRefDataRequestException;
+import com.backend.response.AssetRefDataResponse;
+import com.backend.service.abstractions.IAssetRefDataService;
+
+@RestController
+@RequestMapping(produces = { MediaType.APPLICATION_JSON_VALUE })
+public class AssetRefDataController {
+	Logger logger = LoggerFactory.getLogger(PortfolioController.class);
+    private final IAssetRefDataService assetRefDataService;
+
+    @Autowired
+    public AssetRefDataController(IAssetRefDataService assetRefDataService) {
+        this.assetRefDataService = assetRefDataService;
+    }
+
+	@GetMapping(path = "/asset/{assetId}")
+    public AssetRefDataResponse getAssetRefData(@PathVariable long assetId){
+        List<AssetRefData> assetList = assetRefDataService.findAllByAssetId(assetId);
+        AssetRefDataResponse response = new AssetRefDataResponse();
+        response.setAssetRefDataList(assetList);
+        return response;
+    }
+
+    @GetMapping(path = "/asset/{assetId}/{duration}")
+    public AssetRefDataResponse getHistoricalAssetRefData(@PathVariable("assetId") long assetId, @PathVariable("duration") String duration){
```

Comment:
Should probably use query params instead of path as @DavidLHW mentioned in the previous review! 

Reason is that conventionally path parameter defines the resource location, while the query parameter defines [sort, pagination, or filter](https://www.toolsqa.com/rest-assured/query-parameters-in-rest-assured/) operations. and in this case we want the option to filter the data by duration, hence query params :)